client/src/Components/Layouts/Ordersmanagement/index.jsx

```diff
@@ -1,60 +1,284 @@
 import React, { Component } from "react";
 import axios from "axios";
+import { DatePicker, Input, Button, Icon } from "antd";
+import moment, { isValid } from "moment";
+import Sidebar from "../../CommonComponent/Sidebar/index";
+import Header from "../../CommonComponent/Header/index";
+import Navbar from "../../CommonComponent/Navbar/index";
 import TableComponent from "../../CommonComponent/Table/Table";
+import { EditPopup } from "../../CommonComponent/Table/Popups";
 import {
-  viewPopup,
-  editPopup,
+  //   viewPopup,
   deletePopup
 } from "../../CommonComponent/Table/Popups";
+import "./style.css";
 
 class OrdersManagement extends Component {
   state = {
     orders: [],
-    error: ""
+    filteredOrders: [],
+    date: "",
+    status: "",
+    name: "",
+    error: "",
+    filter: false,
+    stores: []
   };
 
   componentDidMount() {
     axios
       .get("/api/v1//viewOrders")
       .then(res => {
-        this.setState({ orders: res.data });
+        if (res.status === 204) {
+          let error = [...this.state.error];
+          error.response = res;
+          error.response.data = "Error, No orders yet.";
+          this.setState({ error });
+        } else {
+          this.setState({ orders: res.data });
+        }
       })
-      .catch(error => {
+      .catch((error) => {
         this.setState({
           error
         });
       });
+    axios
+      .get("/api/v1/getStores")
+      .then(res => {
+        if (res) {
+          this.setState({ stores: res.data });
+        }
+      })
+      .catch(error => this.setState({ error }));
   }
 
+  dateFilter = object => {
+    if (object) {
+      const { date } = this.state;
+      if (date.length) {
+        if (date[0] && date[1]) {
+          if (date[0].isValid() && date[1].isValid()) {
+            const fromDate = date[0].toDate().setHours(0, 0, 0, 0);
+            const toDate = date[1].toDate().setHours(0, 0, 0, 0);
+            let filtered = object.filter(order => {
+              if (
+                moment(order.date)
+                  .toDate()
+                  .setHours(0, 0, 0, 0) >= fromDate &&
+                moment(order.date)
+                  .toDate()
+                  .setHours(0, 0, 0, 0) <= toDate
+              ) {
+                return true;
+              }
+            });
+            return filtered;
+          }
+        }
+      } else {
+        return object;
```

Comment:
@Anies12 there is an if-statement that doesn't have an else (the one on line 60) so you could probably change this last block, 79-81, to just return (and not be an else statement) and that would handle all of the else conditions.

Reply:
yes it would, thank u